src/reason-parser/reason_template.ml

```diff
@@ -0,0 +1,97 @@
+(**
+ * This module should include utilities for parsing and printing template
+ * strings, but should not have any dependencies on any printing framework
+ * (like Easy_format or Reason_layout). For that, make another module.  This
+ * file should be shared between the printer and the parser, so avoiding
+ * dependencies on printing frameworks, makes it easy to bundle just the parser
+ * if necessary.
+ *)
+
+open Migrate_parsetree
+open Ast_408
+
+module Parse = struct
+  (* Normalizes the last line:
+   *
+   *     `
+   *       abc
+   *       123
+   *       xyz<spc><spc>`
```

Comment:
Is this a typo? It seems like there's one `<spc>` more?

Reply:
No, the first and last white space character is ignored in this non breaking change to Reason Syntax.
In the first breaking change where we change the syntax of polymorphic variants, then we will be able to pay attention to the first and last white space character (that is not a newline). The upgrade process will automatically fix any newlines at the end.

Reply:
Nice. Thanks for the explanation.